airflow/api_fastapi/core_api/routes/public/variables.py

```diff
@@ -58,6 +63,42 @@ async def get_variable(
     return VariableResponse.model_validate(variable, from_attributes=True)
 
 
+@variables_router.get(
+    "/",
+    responses=create_openapi_http_exception_doc([401, 403]),
+)
+async def get_variables(
+    limit: QueryLimit,
+    offset: QueryOffset,
+    order_by: Annotated[
+        SortParam,
+        Depends(
+            SortParam(
+                ["key", "id"],
+                Variable,
+            ).dynamic_depends()
+        ),
+    ],
+    session: Annotated[Session, Depends(get_session)],
+) -> VariableCollectionResponse:
+    """Get all Variables entries."""
+    variable_select, total_entries = paginated_select(
+        select(Variable),
+        [],
+        order_by=order_by,
+        offset=offset,
+        limit=limit,
+        session=session,
+    )
+
+    variables = session.scalars(variable_select).all()
+
+    return VariableCollectionResponse(
+        variables=[VariableResponse.model_validate(variable, from_attributes=True) for variable in variables],
```

Comment:
Are variables appropriately masked?

Reply:
yes they are 
![Screenshot 2024-10-18 at 14 53 14](https://github.com/user-attachments/assets/00c62fd9-2855-4638-a16a-efd62f941e31)

Reply:
Test would be really verbose if asserting the entire payload. But all of this (get_variable, get_variables, etc...) is using the same model which holds that logic and implements it through descriptor. (each get/set to val attribute does it's magic)